src/test/scala/org/constellation/consensus/RoundManagerTest.scala

```diff
@@ -0,0 +1,286 @@
+
+  val readyFacilitators = Map(facilitatorId1 -> peerData1, facilitatorId2 -> peerData2)
+
+  dao.keyPair shouldReturn Fixtures.tempKey
+
+  val tx1 = Fixtures.dummyTx(dao)
+  val tx2 = Fixtures.dummyTx(dao)
+
+  val soe = mock[SignedObservationEdge]
+  soe.baseHash shouldReturn "abc"
+  val tips = (Seq(soe), readyFacilitators)
+
+  dao.id shouldReturn daoId
+  dao.minCheckpointFormationThreshold shouldReturn checkpointFormationThreshold
+  dao.pullTransactions(checkpointFormationThreshold) shouldReturn Some(Seq(tx1, tx2))
+  dao.readyFacilitators() shouldReturn readyFacilitators
+  dao.peerInfo shouldReturn readyFacilitators
+  dao.pullTips(readyFacilitators) shouldReturn Some(tips)
+  dao.threadSafeMessageMemPool shouldReturn mock[ThreadSafeMessageMemPool]
+  dao.threadSafeMessageMemPool.pull(1) shouldReturn None
+  dao.checkpointService shouldReturn mock[CheckpointService]
+  dao.checkpointService.contains(*) shouldReturn true
+  dao.readyPeers(NodeType.Light) shouldReturn Map()
+
+  val peerManagerProbe = TestProbe()
+  val ipManager = mock[IPManager]
+  val peerManager = TestActorRef(Props(new PeerManager(ipManager)))
+  dao.peerManager shouldReturn peerManager
+
+  after {
+    TestKit.shutdownActorSystem(system)
+  }
+
+  test("it should not start a new block-creation round if another round is in progress") {
+    val roundInfo = mock[RoundInfo]
+    roundInfo.startedByThisNode shouldReturn true
+
+    roundManager.underlyingActor.rounds += RoundId("a") -> roundInfo
+
+    roundManager ! StartNewBlockCreationRound
+
+    roundManager.underlyingActor.rounds.size shouldBe 1
+  }
+
+  test("it should start a new block-creation round if another round is not in progress") {
+    roundManager ! StartNewBlockCreationRound
+
+    within(2 seconds) {
+      expectNoMessage
+      roundManager.underlyingActor.rounds.size shouldBe 1
+    }
+  }
+
+  test("it should pass NotifyFacilitators to parent if new round has been started") {
+    roundManager ! StartNewBlockCreationRound
+
+    within(2 seconds) {
+      expectNoMessage
+      roundManager.underlyingActor.passToParentActor(any[NotifyFacilitators]) was called
+    }
+  }
+
+  test("it should pass LightTransactionsProposal to round actor if new round has been started") {
+    roundManager ! StartNewBlockCreationRound
+
+    within(2 seconds) {
+      expectNoMessage
+      roundManager.underlyingActor.passToRoundActor(any[LightTransactionsProposal]) was called
+    }
+  }
+
+  test("it should participate in a new round on ParticipateInBlockCreationRound") {
+    val roundData = new RoundData(
+      RoundId("round1"),
+      Set(peerData1, peerData2),
+      Set(),
+      FacilitatorId(facilitatorId1),
+      Seq(),
+      Seq(),
+      Seq()
+    )
+    val cmd = mock[ParticipateInBlockCreationRound]
+    cmd.roundData shouldReturn roundData
+
+    roundManager ! cmd
+
+    within(2 seconds) {
+      expectNoMessage
+      roundManager.underlyingActor.rounds.size shouldBe 1
```

Comment:
missing startedByThisNode assertion, best would be to match whole RoundInfo if possiblle

Reply:
but then I'll have to know scheduler and actor as well

Reply:
I've added startedByThisNode assertion